File: 30-days-of-code/Day10/Binary.c
/* ************************************************************************** */
/*                                                                            */
/*                                                        :::      ::::::::   */
/*   Binary.c                                           :+:      :+:    :+:   */
/*                                                    +:+ +:+         +:+     */
/*   By: alaamimi <alaamimi@student.42.fr>          +#+  +:+       +#+        */
/*                                                +#+#+#+#+#+   +#+           */
/*   Created: 2021/03/02 09:57:40 by alaamimi          #+#    #+#             */
/*   Updated: 2021/03/02 10:52:53 by alaamimi         ###   ########.fr       */
/*                                                                            */
/* ************************************************************************** */

#include <stdio.h>
#include <math.h>

int	main(void)
{
	int nbr;
	int count;

	count = 0;
	scanf("%d", &nbr);
	while (nbr != 0)
	{
		nbr = nbr & (nbr << 1);
		count++;
	}
	printf("%d", count);
}

File: 30-days-of-code/Day10/binary.py
# **************************************************************************** #
#                                                                              #
#                                                         :::      ::::::::    #
#    binary.py                                          :+:      :+:    :+:    #
#                                                     +:+ +:+         +:+      #
#    By: alaamimi <alaamimi@student.42.fr>          +#+  +:+       +#+         #
#                                                 +#+#+#+#+#+   +#+            #
#    Created: 2021/03/08 17:12:31 by alaamimi          #+#    #+#              #
#    Updated: 2021/03/08 17:14:46 by alaamimi         ###   ########.fr        #
#                                                                              #
# **************************************************************************** #

n = int(input().strip())

current_consecutive_1s = 0
max_consecutive_1s = 0
while n > 0:
    remainder = n % 2
    n = n // 2
    if remainder == 1:
        current_consecutive_1s += 1
        if current_consecutive_1s > max_consecutive_1s:
            max_consecutive_1s = current_consecutive_1s
    else:
        current_consecutive_1s = 0

print(max_consecutive_1s)